builtin/aws/ecs/platform.go

```diff
@@ -1105,11 +1105,21 @@ func (p *Platform) Launch(
 	L.Debug("service started", "arn", servOut.Service.ServiceArn)
 
 	dep := &Deployment{
-		Cluster:         clusterName,
-		TaskArn:         taskArn,
-		ServiceArn:      *servOut.Service.ServiceArn,
-		TargetGroupArn:  *tgArn,
-		LoadBalancerArn: *lb.LoadBalancerArn,
+		Cluster:        clusterName,
+		TaskArn:        taskArn,
+		ServiceArn:     *servOut.Service.ServiceArn,
+		TargetGroupArn: *tgArn,
+	}
+
+	// For deployments without a pre-existing ALB Listener,
+	// we set the LoadBalancerArn directly from the lb found or created.
+	// While for deployments with a pre-existing ALB Listener,
+	// we determine the associated LoadBalancer from the listener
+	// returned from the elbv2.DescribeListeners API response
+	if newListener {
+		dep.LoadBalancerArn = aws.StringValue(lb.LoadBalancerArn)
+	} else {
+		dep.LoadBalancerArn = aws.StringValue(listener.LoadBalancerArn)
 	}
```

Comment:
looking this over again, it seems in the case of no pre-existing ALB listener, the *listener pointer gets set in that branch of the if/else logic so we could maybe simplify this to 
```suggestion
		Cluster:        clusterName,
		TaskArn:        taskArn,
		ServiceArn:     *servOut.Service.ServiceArn,
		TargetGroupArn: *tgArn,
		LoadBalancerArn: aws.StringValue(listener.LoadBalancerArn),
	}
```
🤔 ?